### PR TITLE
refactor: migrate discrete-agent into ts-boilerplate

### PR DESCRIPTION
**Pull request by ronenkapelian**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✔/✖                                                                        |
| New feature     | ✔/✖                                                                        |
| Breaking change | ✔/✖                                                                        |
| Deprecations    | ✔/✖                                                                        |
| Documentation   | ✔/✖                                                                        |
| Tests added     | ✔/✖                                                                        |
| Chore            | ✔/✖                                                                       |

Related issues: #XXX , #XXX ...
Closes #XXX ...

Further  information:

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -4,6 +4,20 @@
   // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
   "version": "0.2.0",
   "configurations": [
+    {
```
Comment (asafMasa) [inline .vscode/launch.json]: revert, run git command to ignore this in the future - [git update-index --assume-unchanged <file>](https://stackoverflow.com/questions/17195861/undo-git-update-index-assume-unchanged-file)

File: config/custom-environment-variables.json [hunk 2]
```diff
@@ -38,5 +38,61 @@
...
+  },
+  "mountDir": "MOUNT_DIRECTORY",
+  "watcher": {
+    "rootDirNestingLevel": {
+      "__name": "ROOT_DIR_NESTING_LEVEL",
+      "__format": "number"
+    },
+    "watchDirectory": "WATCH_DIRECTORY",
+    "watchOptions": {
+      "minTriggerDepth": "WATCH_MIN_TRIGGER_DEPTH",
```
Comment (asafMasa) [inline config/custom-environment-variables.json]: add "__format": "number"

File: config/custom-environment-variables.json [hunk 3]
```diff
@@ -38,5 +38,61 @@
...
+      "maxWatchDepth": "WATCH_MAX_TRIGGER_DEPTH",
```
Comment (asafMasa) [inline config/custom-environment-variables.json]: add "__format": "number"

File: config/custom-environment-variables.json [hunk 4]
```diff
@@ -38,5 +38,61 @@
...
+      "interval": "WATCH_INTERVAL"
```
Comment (asafMasa) [inline config/custom-environment-variables.json]: add "__format": "number"

File: config/custom-environment-variables.json [hunk 5]
```diff
@@ -38,5 +38,61 @@
...
+    },
+    "shpRetry": {
+      "retries": "WATCHER_SHP_RETRY_COUNT",
```
Comment (asafMasa) [inline config/custom-environment-variables.json]: add "__format": "number"

File: config/custom-environment-variables.json [hunk 6]
```diff
@@ -38,5 +38,61 @@
...
+      "factor": "WATCHER_SHP_RETRY_FACTOR",
```
Comment (asafMasa) [inline config/custom-environment-variables.json]: add "__format": "number"

File: config/custom-environment-variables.json [hunk 7]
```diff
@@ -38,5 +38,61 @@
...
+      "minTimeout": "WATCHER_SHP_RETRY_MIN_TIMEOUT",
```
Comment (asafMasa) [inline config/custom-environment-variables.json]: add "__format": "number"

File: config/custom-environment-variables.json [hunk 8]
```diff
@@ -38,5 +38,61 @@
...
+      "maxTimeout": "WATCHER_SHP_RETRY_MAX_TIMEOUT",
```
Comment (asafMasa) [inline config/custom-environment-variables.json]: add "__format": "number"

File: config/custom-environment-variables.json [hunk 9]
```diff
@@ -38,5 +38,61 @@
...
+      "randomize": "WATCHER_SHP_RETRY_RANDOMIZE"
```
Comment (asafMasa) [inline config/custom-environment-variables.json]: add "__format": "boolean"

File: helm/templates/configmap.yaml [hunk 10]
```diff
@@ -1,16 +1,29 @@
...
+{{- $chartName := include "discrete-agent.name" . -}}
+{{- $tracingUrl := include "discrete-agent.tracingUrl" . -}}
+{{- $metricsUrl := include "discrete-agent.metricsUrl" . -}}
 {{- if .Values.enabled -}}
 apiVersion: v1
 kind: ConfigMap
 metadata:
   name: {{ .Release.Name }}-{{ $chartName }}-configmap
 data:
-  REQUEST_PAYLOAD_LIMIT: {{ .Values.env.requestPayloadLimit | quote }}
```
Comment (asafMasa) [inline helm/templates/configmap.yaml:10]: add this config param

File: helm/templates/configmap.yaml [hunk 11]
```diff
@@ -1,16 +1,29 @@
...
-  RESPONSE_COMPRESSION_ENABLED: {{ .Values.env.responseCompressionEnabled | quote }}
```
Comment (asafMasa) [inline helm/templates/configmap.yaml:11]: add this config param

File: helm/templates/configmap.yaml [hunk 12]
```diff
@@ -1,16 +1,29 @@
...
   LOG_LEVEL: {{ .Values.env.logLevel | quote }}
-  LOG_PRETTY_PRINT_ENABLED: {{ .Values.env.logPrettyPrintEnabled | quote }}
```
Comment (asafMasa) [inline helm/templates/configmap.yaml:13]: add this config param

File: helm/values.yaml [hunk 13]
```diff
@@ -16,22 +43,40 @@ cloudProvider:
...
+    accessKeyId: '' # used only when creating secret
+    secretAccessKey: '' # used only when creating secret
+
 env:
-  port: 8080
+  port: 80
   targetPort: 8080
   protocol: TCP
   logLevel: info
-  logPrettyPrintEnabled: false
```
Comment (asafMasa) [inline helm/values.yaml:32]: return this value

File: helm/values.yaml [hunk 14]
```diff
@@ -16,22 +43,40 @@ cloudProvider:
...
-  responseCompressionEnabled: true
```
Comment (asafMasa) [inline helm/values.yaml:33]: return this value

File: helm/values.yaml [hunk 15]
```diff
@@ -16,22 +43,40 @@ cloudProvider:
...
-  requestPayloadLimit: 1mb
```
Comment (asafMasa) [inline helm/values.yaml:34]: return this value

File: src/layerCreator/models/shpParser.ts [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+  }
+
+  private async toGeoJson(shp: string, dbf: string, encoding = 'utf-8'): Promise<GeoJSON> {
+    try {
+      return await readShp(shp, dbf, { encoding: encoding });
+    } catch (err) {
+      const error = err as Error;
+      this.logger.error(`failed to parse shapeFile: ${error.message}`);
+      throw new BadRequestError(error, `Invalid shp file: ${shp}`);
+      // throw new BadRequestError(`Invalid shp file: ${shp}, err: ${error.message}`);
```
Comment (asafMasa) [inline src/layerCreator/models/shpParser.ts]: remove remark

File: src/serviceClients/agentDbClient.ts [hunk 17]
```diff
@@ -0,0 +1,86 @@
...
+    } catch (err) {
+      const error = err as Error;
+      this.logger.error(`failed to retrieve watch status from db, error=${error.message}`);
+      throw err;
+    }
+  }
+
+  public async setWatchStatus(status: IWatchStatus): Promise<IWatchStatus> {
+    try {
+      return await this.put('/status', status);
```
Comment (asafMasa) [inline src/serviceClients/agentDbClient.ts:80]: add info log

**Review by asafMasa — CHANGES_REQUESTED** [review]
Good Job👍

File: src/serviceClients/agentDbClient.ts (hunk 17 above)
Comment (ronenkapelian) [reply src/serviceClients/agentDbClient.ts:80]: done

File: src/layerCreator/models/shpParser.ts (hunk 16 above)
Comment (ronenkapelian) [reply src/layerCreator/models/shpParser.ts]: done

File: helm/values.yaml (hunk 15 above)
Comment (ronenkapelian) [reply helm/values.yaml:34]: done

File: helm/values.yaml (hunk 14 above)
Comment (ronenkapelian) [reply helm/values.yaml:33]: done

File: helm/values.yaml (hunk 13 above)
Comment (ronenkapelian) [reply helm/values.yaml:32]: done

File: config/custom-environment-variables.json (hunk 5 above)
Comment (ronenkapelian) [reply config/custom-environment-variables.json]: done

File: config/custom-environment-variables.json (hunk 4 above)
Comment (ronenkapelian) [reply config/custom-environment-variables.json]: done

File: config/custom-environment-variables.json (hunk 3 above)
Comment (ronenkapelian) [reply config/custom-environment-variables.json]: done

File: config/custom-environment-variables.json (hunk 2 above)
Comment (ronenkapelian) [reply config/custom-environment-variables.json]: done

File: .vscode/launch.json (hunk 1 above)
Comment (ronenkapelian) [reply .vscode/launch.json]: done

File: config/custom-environment-variables.json [hunk 18]
```diff
@@ -45,18 +45,45 @@
       "__name": "ROOT_DIR_NESTING_LEVEL",
       "__format": "number"
     },
-    "watchDirectory": "WATCH_DIRECTORY",
+    "watchDirectory": {
+      "__name": "WATCH_DIRECTORY",
+      "__format": "number"
```
Comment (asafMasa) [inline config/custom-environment-variables.json]: bug!

**Review by asafMasa — CHANGES_REQUESTED** [review]
You didn't resolve all of the previous comments

File: config/custom-environment-variables.json (hunk 18 above)
Comment (ronenkapelian) [reply config/custom-environment-variables.json]: done, fixed.

File: helm/templates/configmap.yaml (hunk 10 above)
Comment (ronenkapelian) [reply helm/templates/configmap.yaml:10]: done

File: helm/templates/configmap.yaml (hunk 11 above)
Comment (ronenkapelian) [reply helm/templates/configmap.yaml:11]: done

File: helm/templates/configmap.yaml (hunk 12 above)
Comment (ronenkapelian) [reply helm/templates/configmap.yaml:13]: done

**Review by asafMasa — APPROVED** [review]

**Pull request merged**